ST_seedDispersal.c

```diff
@@ -0,0 +1,325 @@
+/*********************************************************************************
+ * ST_seedDispersal.c
+ * 
+ * Contains definitions for all functions belonging to the seed dispersal module.
+ *********************************************************************************/
+
+#include "ST_globals.h"
+#include "ST_defines.h"
+#include "ST_grid.h"
+#include "ST_seedDispersal.h"
+#include "rands.h"
+#include "myMemory.h"
+
+int _do_bulk_dispersal(SppIndex sp);
+void _do_precise_dispersal(int leftoverSeeds, SppIndex sp);
+float _cell_dist(int row1, int row2, int col1, int col2, float cellLen);
+
+/* RNG unique to seed dispersal. */
+pcg32_random_t dispersal_rng;
+
+/* Derives the probabilities that a given cell will disperse seeds to any other cell. 
+   results of this function can be accessed using 
+   gridCells[a][b].mySeedDispersal[s].dispersalProb[c][d] 
+   where (a,b) are the coordinates of the sender,
+   (b, c) are the coordinates of the reciever,
+   and s is the species. */
+void initDispersalParameters(void)
+{
+	int senderRow, senderCol, recieverRow, recieverCol;
+	SppIndex sp;
+	double MAXD; /* maximum seed dispersal distance for a given species */
+	double maxRate; /* Dispersability of the seeds */
+	double plotWidth; /* width of the plots (all plots are equal and square) */
+	double distanceBetweenPlots; /* distance between the sender and the reciever */
+    CellType* sender;
+
+    RandSeed(SuperGlobals.randseed, &dispersal_rng);
+
+	/* sender denotes that these loops refer to the cell distributing seeds */
+	for(senderRow = 0; senderCol < grid_Rows; ++senderRow){
+		for(senderCol = 0; senderCol < grid_Cols; ++senderCol){
+			/* Cell is loaded to ensure the ForEachSpecies loop works */
+			load_cell(senderRow, senderCol);
+            sender = &gridCells[senderRow][senderCol];
+
+			/* Allocate seed dispersal information. */
+			sender->mySeedDispersal = Mem_Calloc(MAX_SPECIES, sizeof(Grid_SD_St), "initDispersalParameters");
+
+			ForEachSpecies(sp){
+				if (!(Species[sp]->use_me && Species[sp]->use_dispersal)){
```

Comment:
We need the second statement because some species could be on for the simulation but off for seed dispersal. If we removed the second condition it would be impossible to turn off seed dispersal for a specific species.

Reply:
@chaukap thanks for the clarification!